src/%IPM/Utils/Module.cls

```diff
@@ -1980,4 +2110,90 @@ ClassMethod GetFileLines(pFileName As %String, Output pOutput) As %Status [ Inte
+ClassMethod %EvaluateSystemExpression(pString As %String) As %String [ Internal ]
+{
+	If (pString '[ "{") {
+		Return pString
+	}
+	Set result = pString
+	Do ..GetDatabaseInfoForNamespace($Namespace, .properties)
+	If $Locate(result,"(?i)\{\$namespace\}|\$\{namespace\}") {
+		Set result = ..%RegExReplace(result,"(?i)\{\$namespace\}|\$\{namespace\}",$Namespace)
+	}
+	If $Locate(result,"(?i)\{\$ns\}|\$\{ns\}") {
+		Set result = ..%RegExReplace(result,"(?i)\{\$ns\}|\$\{ns\}",$Namespace)
+	}
+	If $Locate(result,"(?i)\{\$namespaceLower\}|\$\{namespaceLower\}") {
+		Set result = ..%RegExReplace(result,"(?i)\{\$namespaceLower\}|\$\{namespaceLower\}",$ZConvert($Namespace,"L"))
+	}
+	If $Locate(result,"(?i)\{\$namespaceRoutineDB\}|\$\{namespaceRoutineDB\}") {
+		Set result = ..%RegExReplace(result,"(?i)\{\$namespaceRoutineDB\}|\$\{namespaceRoutineDB\}",$Get(properties("Routines")))
+	}
+	If $Locate(result,"(?i)\{\$namespaceGlobalsDB\}|\$\{namespaceGlobalsDB\}") {
+		Set result = ..%RegExReplace(result,"(?i)\{\$namespaceGlobalsDB\}|\$\{namespaceGlobalsDB\}",$Get(properties("Globals")))
+	}
+	If $Locate(result,"(?i)\{\$installDir\}|\$\{installDir\}") {
+		Set dir = $System.Util.InstallDirectory()
+		Set result = ..%RegExReplace(result,"(?i)\{\$installDir\}|\$\{installDir\}",dir)
+	}
+	If $Locate(result,"(?i)\{\$dataDir\}|\$\{dataDir\}") {
+		Set dir = $System.Util.DataDirectory()
+		Set result = ..%RegExReplace(result,"(?i)\{\$dataDir\}|\$\{dataDir\}",dir)
+	}
+	If $Locate(result,"(?i)\{\$mgrDir\}|\$\{mgrDir\}") {
+		Set dir = ##class(%Library.File).ManagerDirectory()
+		Set result = ..%RegExReplace(result,"(?i)\{\$mgrDir\}|\$\{mgrDir\}",dir)
+	}
+	If $Locate(result,"(?i)\{\$cspDir\}|\$\{cspDir\}") {
+		Set dir = ##class(%Library.File).NormalizeDirectory($System.Util.InstallDirectory()_"csp")
+		Set result = ..%RegExReplace(result,"(?i)\{\$cspDir\}|\$\{cspDir\}",dir)
+	}
+	If $Locate(result,"(?i)\{\$binDir\}|\$\{binDir\}") {
+		Set dir = $System.Util.BinaryDirectory()
+		Set result = ..%RegExReplace(result,"(?i)\{\$binDir\}|\$\{binDir\}",dir)
+	}
+	If $Locate(result,"(?i)\{\$libDir\}|\$\{libDir\}") {
+		Set dir = ##class(%Library.File).NormalizeDirectory($System.Util.InstallDirectory()_"lib")
+		Set result = ..%RegExReplace(result,"(?i)\{\$libDir\}|\$\{libDir\}",dir)
+	}
+	If $Locate(result,"(?i)\{\$webroot\}|\$\{webroot\}") {
+		Do ##class(%Studio.General).GetWebServerPort(,,,.urlRoot)
+		Set result = ..%RegExReplace(result,"(?i)\{\$webroot\}|\$\{webroot\}",urlRoot)
+	}
+	Return result
+}
+
+/// Wrapper around Config.Namespaces:Get()
+ClassMethod GetDatabaseInfoForNamespace(pNamespace As %String, Output pProperties) [ Internal ]
+{
+	Kill pProperties
+	New $Namespace
+	Set $Namespace = "%SYS"
+	Set sc = $ClassMethod("Config.Namespaces", "Get", pNamespace, .pProperties)
+}
+
+ClassMethod %RegExReplace(pString, pRegEx, pValue) As %String [ Internal ]
+{
+	Quit ##class(%Regex.Matcher).%New(pRegEx,pString).ReplaceAll($Replace(pValue,"\","\\"))
+}
```

Comment:
Move this method from %IPM.Storage.Module
That method is universal for both `${var}` and `{$var}`

And move variables valuation from that class
and the code will be more readable, as well, as less possible errors
```
	Set tAttrValue = ..%RegExReplace(tAttrValue, "namespace", $Namespace)
	Set tAttrValue = ..%RegExReplace(tAttrValue, "ns",        $Namespace)
	Set tAttrValue = ..%RegExReplace(tAttrValue, "mgrdir",    tMgrDir)
	Set tAttrValue = ..%RegExReplace(tAttrValue, "cspdir",    tCSPDir)
	Set tAttrValue = ..%RegExReplace(tAttrValue, "root",      tRoot)
	Set tAttrValue = ..%RegExReplace(tAttrValue, "bindir",    tBinDir)
	Set tAttrValue = ..%RegExReplace(tAttrValue, "libdir",    tLibDir)
	Set tAttrValue = ..%RegExReplace(tAttrValue, "verbose",   tVerbose)
	Set tAttrValue = ..%RegExReplace(tAttrValue, "webroot",   urlRoot)
```

Reply:
@daimor  Comments resolved. Also added the new properties to handle display in case-sensitive (storage will be in all lower case as before)